### PR TITLE
Target dotnet TFM's for class libraries

### PR DESCRIPTION
**Pull request by henkmollema**
- Target `net451` and `dotnet5.4`
- Add SPA services project to solution (with .xproj)
- Update .gitignore
- Add Git repo information to project.json

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,31 @@
...
+*.ncrunchsolution
+*.*sdf
+*.ipch
+*.sln.ide
+node_modules
+*.sln.ide
 project.lock.json
-npm-debug.log
+.vs/
+npm-debug.log
```
Comment (SteveSandersonMS) [inline .gitignore]: About this file - it would be nice to keep it minimal. Could we limit these entries just to the files that appear without the developer's involvement on the two most common dev setups: VS and VS Code?

File: Microsoft.AspNet.AngularServices/project.json [hunk 2]
```diff
@@ -1,20 +1,16 @@
 {
   "version": "1.0.0-alpha7",
-  "description": "Microsoft.AspNet.AngularServices Class Library",
-  "authors": [
-    "Microsoft"
```
Comment (SteveSandersonMS) [inline Microsoft.AspNet.AngularServices/project.json]: This information seems to get lost. Can we retain it?

File: Microsoft.AspNet.AngularServices/project.json [hunk 3]
```diff
@@ -1,20 +1,16 @@
...
-  ],
-  "tags": [
-    ""
-  ],
-  "projectUrl": "",
-  "licenseUrl": "",
+  "description": "ASP.NET 5 server-side Angular services.",
```
Comment (SteveSandersonMS) [inline Microsoft.AspNet.AngularServices/project.json]: It also contains client-side stuff. Maybe rephrase to "Helpers for building Angular 2 applications on ASP.NET 5"

File: Microsoft.AspNet.NodeServices/project.json [hunk 4]
```diff
@@ -1,36 +1,32 @@
 {
   "version": "1.0.0-alpha7",
-  "description": "Microsoft.AspNet.NodeServices",
-  "authors": [
-    "Microsoft"
```
Comment (SteveSandersonMS) [inline Microsoft.AspNet.NodeServices/project.json]: This information seems to get lost. Can we retain it?

File: Microsoft.AspNet.NodeServices/project.json [hunk 5]
```diff
@@ -1,36 +1,32 @@
...
-  ],
-  "tags": [
-    ""
-  ],
-  "projectUrl": "",
-  "licenseUrl": "",
+  "description": "ASP.NET 5 Node.js services.",
```
Comment (SteveSandersonMS) [inline Microsoft.AspNet.NodeServices/project.json]: Maybe "Invoke Node.js modules at runtime in ASP.NET 5 applications"

File: Microsoft.AspNet.ReactServices/project.json [hunk 6]
```diff
@@ -1,20 +1,21 @@
 {
   "version": "1.0.0-alpha7",
-  "description": "Microsoft.AspNet.ReactServices Class Library",
-  "authors": [
-    "Microsoft"
```
Comment (SteveSandersonMS) [inline Microsoft.AspNet.ReactServices/project.json]: This information seems to get lost. Can we retain it?

File: Microsoft.AspNet.ReactServices/project.json [hunk 7]
```diff
@@ -1,20 +1,21 @@
...
-  ],
-  "tags": [
-    ""
-  ],
-  "projectUrl": "",
-  "licenseUrl": "",
+  "description": "ASP.NET 5 server-side React services,",
```
Comment (SteveSandersonMS) [inline Microsoft.AspNet.ReactServices/project.json]: Maybe rephrase to "Helpers for building React applications on ASP.NET 5"

File: NuGet.config [hunk 8]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (SteveSandersonMS) [inline NuGet.config]: Could you educate me on what the role of this file is? What problem occurs if it's not present?

Comment (henkmollema) [inline NuGet.config]: It configures the NuGet feeds used by DNX/DNU. Makes it easier to swap out a certain feeds, rather than using the global NuGet.config.

Other repo's use it as well.

Comment (SteveSandersonMS) [inline NuGet.config]: OK, thanks for clarifying. I wondered if it was strictly necessary in some situation. If a scenario occurs where this is needed then we can add it. Until then I'm trying to minimise the amount of boilerplate stuff.

Comment (henkmollema) [inline NuGet.config]: I think CI needs it too, if you want that.

Comment (SteveSandersonMS) [inline NuGet.config]: OK that's fine. I'll add it then if needed :)

Thanks for this pull request BTW.

Comment (henkmollema) [inline NuGet.config]: :+1:

**Comment by henkmollema** [comment]
@SteveSandersonMS updated

**Comment by SteveSandersonMS** [comment]
Thanks! It's in.

**Comment by henkmollema** [comment]
Cool! :smile:
